internal/provider/resource_user.go

```diff
@@ -170,6 +177,13 @@ func resourceUserUpdate(ctx context.Context, d *schema.ResourceData, meta interf
 		return diag.FromErr(err)
 	}
 
+	if d.HasChange("pin") {
+		if v, ok := d.GetOk("pin"); ok {
+			user.Pin.IsSet = false
```

Comment:
Looks like no one read `IsSet` property, why do you reset it? Probably I missed something?

Reply:
This was a relic from an earlier iteration. Removed it.